src/backend/dx11/src/device.rs

```diff
@@ -919,43 +919,62 @@ impl device::Device<Backend> for Device {
                 .map_err(|err| pso::CreationError::Shader(err))
         };
 
-        let vs = build_shader(pso::Stage::Vertex, Some(&desc.shaders.vertex))?.unwrap();
-        let ps = build_shader(pso::Stage::Fragment, desc.shaders.fragment.as_ref())?;
-        let gs = build_shader(pso::Stage::Geometry, desc.shaders.geometry.as_ref())?;
-        let ds = build_shader(pso::Stage::Domain, desc.shaders.domain.as_ref())?;
-        let hs = build_shader(pso::Stage::Hull, desc.shaders.hull.as_ref())?;
-
-        let layout = self.create_input_layout(
-            vs.clone(),
-            &desc.vertex_buffers,
-            &desc.attributes,
-            &desc.input_assembler,
-        )?;
-        let rasterizer_state = self.create_rasterizer_state(&desc.rasterizer)?;
-        let blend_state = self.create_blend_state(&desc.blender)?;
-        let depth_stencil_state = Some(self.create_depth_stencil_state(&desc.depth_stencil)?);
+        let (layout, vs, gs, hs, ds) = match desc.primitive_assembler {
+            pso::PrimitiveAssembler::Vertex {
+                ref buffers,
+                ref attributes,
+                ref input_assembler,
+                ref vertex,
+                ref tessellation,
+                ref geometry,
+            } => {
+                let (hs, ds) = if let Some(ts) = tessellation {
+                    (Some(&ts.0), Some(&ts.1))
```

Comment:
looks like we could have combined this match with the later one

Reply:
Done